Correctly set author_key_digest and report_id fields in js_verify_snp_attestation

achamayou opened the pull request:
Fields were incorrectly set due to a presumed copy-paste mistake.

maxtropets commented:
Wonder why never caught with smth like use-after-move warning

achamayou commented:
@maxtropets that's a good question, because as far I understand it, we enable bugprone-* checks in clang-tidy: https://github.com/microsoft/CCF/blob/410a993fec27da4285a2ecb20aef35aef92f3471/.clang-tidy#L4, which should cover https://clang.llvm.org/extra/clang-tidy/checks/bugprone/use-after-move.html

eddyashton commented:
> @maxtropets that's a good question, because as far I understand it, we enable bugprone-* checks in clang-tidy:
> 
> https://github.com/microsoft/CCF/blob/410a993fec27da4285a2ecb20aef35aef92f3471/.clang-tidy#L4
> 
> , which should cover https://clang.llvm.org/extra/clang-tidy/checks/bugprone/use-after-move.html

This is very weird, and worth a little more investigation? Possible explanations:
- `clang-tidy` isn't running on this code
- This check doesn't consider another "move" to be an additional "use" (ie `move(x); y=x+1` is obviously wrong, but `move(x); move(x)` _might_ be safe?)
- This check can't handle the fact we're making this call inside a macro (which _could_ interfere with comma expansion, I guess, but drastically reduces the coverage we're actually getting from `clang-tidy`)

achamayou commented:
@eddyashton it doesn't seem to be option 2, a trivial move after move is definitely noisily caught:

```
    std::string foo = "Hello, World!";
    std::string bar = std::move(foo);
    std::string baz = std::move(foo);
```

```
$ clang-tidy --checks=-bugprone-unused-local-non-trivial-variable test.cpp 
...
/home/amchamay/CCF/test.cpp:8:23: error: Moved-from object 'foo' of type 'std::basic_string' is moved [clang-analyzer-cplusplus.Move,-warnings-as-errors]
    8 |     std::string baz = std::move(foo);
      |                       ^~~~~~~~~~~~~~
/home/amchamay/CCF/test.cpp:7:23: note: Object 'foo' of type 'std::basic_string' is left in a valid but unspecified state after move
    7 |     std::string bar = std::move(foo);
      |                       ^~~~~~~~~~~~~~
/home/amchamay/CCF/test.cpp:8:23: note: Moved-from object 'foo' of type 'std::basic_string' is moved
    8 |     std::string baz = std::move(foo);
      |                       ^~~~~~~~~~~~~~
/home/amchamay/CCF/test.cpp:8:33: error: 'foo' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
    8 |     std::string baz = std::move(foo);
      |                                 ^
/home/amchamay/CCF/test.cpp:7:23: note: move occurred here
    7 |     std::string bar = std::move(foo);
      |                       ^
```

achamayou commented:
Well duh, it's option 1:  https://github.com/microsoft/CCF/blob/main/CMakeLists.txt#L426
